classes/class08/README.md

```diff
@@ -1 +1,101 @@
+
+This is all great for synchronous testing, but how do we deal with asynchronicity? We need some way to tell mocha to wait for our callbacks to run and assertions in them to fire before it finishes testing. We can do that in our call to `it` when setting up our tests. The second argument to `it` is a function, and that function can have 0 or 1 arguments. In the 0 argument case, it does synchronous tests like above. In the 1 argument case, however, your `it` function receives as an argument another function, idiomatically called `done`. Now, mocha will wait until `done` is called before it proceeds with more testing, allowing you to call `done` after running assertions in a callback to run async tests. Uncomment lines 15-21 and you can see this in action. Running our tests now shows 2 passing, one of which takes about a second, showing that our async test is working. Async tests aren't that complicated to do in mocha, but they are a little dangerous. If `done` never gets called for some reason (like an error in your callback), mocha will not know to stop waiting for it, and your tests will hang. To prevent haning, mocha has a default timeout of 2 seconds on async tests, after which they will fail
+
+##Running Tasks with `npm`
+
+We can configure `npm` to do quite a bit more than just manage packages. In your `package.json`, you can specify scripts that `npm` can then run.
+
+###### package.json
+```
+{
+  ...
+  "main": "index.js",
+  "scripts": {
+  	"hello": "echo hi"
+  }
+  ...
+}
+```
+
+Run scripts with `npm run scriptname`
+
+```bash
+$ npm run hello
+hi
+```
+
+Below is the `scripts` section from the in-class exercise `package.json` showing how we can use `npm` to run our unit tests. As you can see, you can even run `npm` tasks with `npm`!
+
+###### package.json
+```
+{
+  ...
+  "main": "index.js",
+  "scripts": {
+    "karma": "./node_modules/karma/bin/karma start karma.conf.js",
+    "mocha": "./node_modules/mocha/bin/mocha tests/server",
+    "cover-mocha": "./node_modules/istanbul/lib/cli.js cover ./node_modules/mocha/bin/_mocha tests/server -- -R spec",
+    "test": "npm run karma && npm run cover-mocha",
```

Comment:
You should note that `npm test` and `npm start` do not require the `run` syntax.